### DISCUSSION TITLE
Why did MPL-2.0 start to creep in?

### DISCUSSION THREAD

**Discussion by vazub** [yshui/picom#888] [Q&A] [answered]
This appears to be a bit shady case of re-licensing of a perfectly permissive code. Sure, it is allowed by the MIT license itself, but still is a rather bad move, especially towards OSes that try to avoid copyleft software or otherwise isolate its use. Is there any rationale published in this regard?

**Comment by yshui** [comment, marked as answer]
So to clarify: nothing is being re-licensed, it's just any new code I add, especially in new source files, is licensed under MPL-2.

And the reason is simple, and is the likely most common reason why people choose a copyleft license: I don't want others to take my code and benefit from it without contributing back.
